GDJS/Runtime/runtimeobject.ts

```diff
@@ -2330,16 +2339,14 @@ namespace gdjs {
         if (res.collision) {
           if (closest && res.closeSqDist < testSqDist) {
             testSqDist = res.closeSqDist;
-            result = res;
-          } else {
-            if (
-              !closest &&
-              res.farSqDist > testSqDist &&
-              res.farSqDist <= raySqBoundingRadius
-            ) {
-              testSqDist = res.farSqDist;
-              result = res;
-            }
+            gdjs.Polygon.copyRaycastTestResult(res, result);
+          } else if (
+            !closest &&
+            res.farSqDist > testSqDist &&
+            res.farSqDist <= raySqBoundingRadius
+          ) {
+            testSqDist = res.farSqDist;
+            gdjs.Polygon.copyRaycastTestResult(res, result);
           }
```

Comment:
The fix is to avoid to override previous results during the loop, right?

I have only some suggestion about readability.

Rename `closest` to `shouldSearchClosest`?
`testSqDist` could be split in `sqDistMin` and `sqDistMax` for clarity

A if-else could better show that only one of the 2 blocks can be executed in one loop. I would even prefer the loop to be inside the if-else.
```JavaScript
if (closest) {
  if ()
}
else {
  if ()
}
```

Reply:
> The fix is to avoid to override previous results during the loop, right?

Yes!

> I have only some suggestions about readability.

Fair enough, I see what you're saying! I'll give it a try.